src/Microsoft.ML.PipelineInference/AutoMlEngines/RocketEngine.cs

```diff
@@ -196,21 +196,10 @@ private void SampleHyperparameters(RecipeInference.SuggestedRecipe.SuggestedLear
         private RecipeInference.SuggestedRecipe.SuggestedLearner[] GetTopLearners(IEnumerable<PipelinePattern> history)
         {
             var weights = LearnerHistoryToWeights(history.ToArray(), IsMaximizingMetric);
-            var topKTuples = new Tuple<double, int>[_topK];
-
-            for (int i = 0; i < weights.Length; i++)
-            {
-                if (i < _topK)
-                    topKTuples[i] = new Tuple<double, int>(weights[i], i);
-                else
-                {
-                    for (int j = 0; j < topKTuples.Length; j++)
-                        if (weights[i] > topKTuples[j].Item1)
-                            topKTuples[j] = new Tuple<double, int>(weights[i], i);
-                }
-            }
-
-            return topKTuples.Select(t => AvailableLearners[t.Item2]).ToArray();
+            return weights.Select((w, i) => new { Weight = w, Index = i })
+                .OrderByDescending(x => x.Weight)
+                .Take(_topK)
+                .Select(t=>AvailableLearners[t.Index]).ToArray();
```

Comment:
>t=>Av [](start = 24, length = 5)

ctrl-k-d to keep formatting up to date and whatnot

Reply:
Loading the file into Visual Studio and pressing ctrl+k+d has no effect.  (I'm guessing that is already the "preferred" indentation?)

Reply:
Tom probably talks about t=>AvailableLearners and want it to look like t => AvailableLearners.
My VS btw put spaces, maybe setting for that on your machine is disabled.

---
In reply to: [191921117](https://github.com/dotnet/machinelearning/pull/270#discussion_r191921117) [](ancestors = 191921117)